internal/utils/status/status.go

```diff
@@ -86,10 +86,10 @@ type ResourceConditions struct {
 // checked by the caller to prevent reconciliation loops.
 func GetResourceConditions(
 	ctx context.Context, namespace string, dynamicClient dynamic.Interface,
-	gvr schema.GroupVersionResource, labelSelector string,
+	gvr schema.GroupVersionResource, labelSelector string, limit int,
```

Comment:
I don't quite understand the purpose of the `limit` param. Seem like it could be omitted.